docs/README.md

```diff
@@ -0,0 +1,135 @@
+# Rufio
+## Whos is Rufio?
+Besides being the leader of the Lost Boys after Peter Pan left Neverland, Rufio is a kubernetes controller for managing baseboard management controllers (BMC) in a Tinkerbell context. Rufio can also execute jobs to perform a set of one-off management actions to bring a machine (physical hardware) to a desired state.
+## Architecture
+![architecture](puml/architecture.png)
+Rufio controller consists of three main API types, [Machine](https://github.com/tinkerbell/rufio/blob/main/api/v1alpha1/machine.go), [Job](https://github.com/tinkerbell/rufio/blob/main/api/v1alpha1/job.go) and [Task](https://github.com/tinkerbell/rufio/blob/main/api/v1alpha1/task.go). An operator or an automated client like [CAPT](https://github.com/tinkerbell/cluster-api-provider-tinkerbell) can interact with Rufio using these APIs to manage the state of their physical machines.
+### Machine API
+The Machine type contains the information required for communicating with the BMC of the physical hardware.
+```yaml
+apiVersion: bmc.tinkerbell.org/v1alpha1
+kind: Machine
+metadata:
+  name: machine-sample
+spec:
+  connection:
+    host: 0.0.0.0
+    port: 623
+    authSecretRef:
+      name: bm-auth
+      namespace: sample
+    insecureTLS: false
+```
+The `connection` object contains the required fields for establising a BMC connection. Fields `host`, `port` represent the BMC IP for the physical machine and `insecureTLS` instructs weather to use insecure TLS connectivity for performing BMC API calls. Field `authSecretRef` is a `SecretReference` which points to a kubernetes secret that contains the username/password for authenticating BMC API calls.
+### Machine controller
+When a Machine object is created on the cluster, the machine controller is responsible for updating the current state of the physical machine. It performs redfish/ipmitool API calls to the BMC on the physical machine and updates the `status` of the Machine object.
```

Comment:
bmclib abstracts away the protocols used to interact with a BMC. might think about making this statement (`redfish/ipmitool API calls`) more generic.

Reply:
Good point. Will remove that.